### PR TITLE
[consensus_adapter] Count and limit number of in-flight transactions submitted to narwhal

### PR DESCRIPTION
**Pull request by andll**
This PR also removes annoying Option<Metrics> pattern from consensus adapter (planned it long time ago)

### CODE REVIEW

File: crates/sui-core/src/consensus_adapter.rs [hunk 1]
```diff
@@ -125,11 +134,23 @@ impl ConsensusAdapterMetrics {
                 authority_position_buckets.to_vec(),
                 registry,
             ).unwrap(),
-        }))
+            in_flight_semaphore_wait: register_int_gauge_with_registry!(
+                "in_flight_semaphore_wait",
```
Comment (mwtian) [inline crates/sui-core/src/consensus_adapter.rs:138]: Should we give this metric a prefix, e.g. `consensus_adapter_` or `sequencing_`?

File: crates/mysten-metrics/src/guards.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+}
+
+impl<'a> Drop for GaugeGuard<'a> {
+    fn drop(&mut self) {
+        self.0.dec();
+    }
+}
+
+pub trait GaugeGuardFutureExt: Future + Sized {
```
Comment (mwtian) [inline crates/mysten-metrics/src/guards.rs:25]: Add a comment for this as well?

**Review by mwtian — APPROVED** [review]

File: crates/sui-core/src/consensus_adapter.rs (hunk 1 above)
Comment (andll) [reply crates/sui-core/src/consensus_adapter.rs:138]: I'll add `consensus_` prefix

Comment (andll) [reply crates/sui-core/src/consensus_adapter.rs:138]: Actually you are right, `sequencing_` seem to be more common

File: crates/mysten-metrics/src/guards.rs (hunk 2 above)
Comment (andll) [reply crates/mysten-metrics/src/guards.rs:25]: Done(added comment to the count_in_flight method)

**Pull request merged**